cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -959,6 +959,10 @@ KernelType GenerateNumeric(detail::GetTypeId get_id) {
       return Generator<Type0, FloatType, Args...>::Exec;
     case Type::DOUBLE:
       return Generator<Type0, DoubleType, Args...>::Exec;
+    case Type::HALF_FLOAT:
+      // NOTE: Type::HALF_FLOAT used to not be part of the list of numeric types,
+      // so users of this template might start failing to compiler after Arrow 17.x.
```

Comment:
These are internal helpers, so if Arrow still compiles, I don't think the comment is warranted.

Reply:
Cool. It took me a while to realize these were internal. I will remove the comments.